src/auth/user-record.ts

```diff
@@ -57,19 +58,226 @@ export interface CreateRequest extends UpdateRequest {
+  lastLoginAt?: string;
+}
+
+export interface MfaInfo {
+  mfaEnrollmentId: string;
+  displayName?: string;
+  phoneInfo?: string;
+  enrolledAt?: string;
+  [key: string]: any;
+}
+
+export interface ProviderUserInfo {
+  rawId: string;
+  displayName?: string;
+  email?: string;
+  photoUrl?: string;
+  phoneNumber?: string;
+  providerId: string;
+  federatedId?: string;
+}
+
+export interface Mfa {
+  mfaInfo?: MfaInfo[];
+}
+
+export interface GetAccountInfoUserResponse extends Metadata, Mfa {
+  localId: string;
+  email?: string;
+  emailVerified?: boolean;
+  phoneNumber?: string;
+  displayName?: string;
+  photoUrl?: string;
+  disabled?: boolean;
+  passwordHash?: string;
+  salt?: string;
+  customAttributes?: string;
+  validSince?: string;
+  tenantId?: string;
+  providerUserInfo?: ProviderUserInfo[];
+  [key: string]: any;
+}
+
+/** Enums for multi-factor identifiers. */
+export enum MultiFactorId {
+  Phone = 'phone',
+}
+
+/**
+ * Abstract class representing a multi-factor info interface.
+ */
+export abstract class MultiFactorInfo {
+  public readonly uid: string;
+  public readonly displayName: string | null;
+  public readonly factorId: MultiFactorId;
+  public readonly enrollmentTime: string;
+
+  /**
+   * Initializes the MultiFactorInfo associated subclass using the server side.
+   * If no MultiFactorInfo is associated with the response, null is returned.
+   *
+   * @param response The server side response.
+   * @constructor
+   */
+  public static initMultiFactorInfo(response: MfaInfo): MultiFactorInfo | null {
+    let multiFactorInfo: MultiFactorInfo | null = null;
+    // Only PhoneMultiFactorInfo currently available.
+    try {
+      multiFactorInfo = new PhoneMultiFactorInfo(response);
+    } catch (e) {
+      // Ignore error.
+    }
+    return multiFactorInfo;
+  }
+
+  /**
+   * Initializes the MultiFactorInfo object using the server side response.
+   *
+   * @param response The server side response.
+   * @constructor
+   */
+  constructor(response: MfaInfo) {
+    const factorId = this.getFactorId(response);
+    this.initFromServerResponse(response, factorId);
+  }
+
+  /** @return The plain object representation. */
+  public toJSON(): any {
+    return {
+      uid: this.uid,
+      displayName: this.displayName,
+      factorId: this.factorId,
+      enrollmentTime: this.enrollmentTime,
+    };
+  }
+
+  /**
+   * Returns the factor ID based on the response provided.
+   *
+   * @param response The server side response.
+   * @return The multi-factor ID associated with the provided response. If the response is
+   *     not associated with any known multi-factor ID, null is returned.
+   */
+  protected abstract getFactorId(response: MfaInfo): MultiFactorId | null;
+
+  /**
+   * Initializes the MultiFactorInfo object using the provided server response.
+   *
+   * @param response The server side response.
+   */
+  private initFromServerResponse(response: MfaInfo, factorId: MultiFactorId | null) {
```

Comment:
You can call `getFactorId()` in this method, and remove the `factorId` argument.

Reply:
Done.